src/common/Tile/GenreSection/index.js

```diff
@@ -1,12 +1,18 @@
 import React from "react";
-import { useSelector } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import GenreTile from "../GenreTile";
 import { StyledSection } from "./styled";
-import { selectGenres } from "../../../features/genres/genresSlice";
+import { fetchGenres, selectGenres } from "../../../features/genres/genresSlice";
+import { useEffect } from "react";
 
 const GenreSection = ({ horizontal, genres }) => {
+  const dispatch = useDispatch();
   const genresList = useSelector(selectGenres);
 
+  useEffect(() => {
+    !genresList.genres && dispatch(fetchGenres());
+  }, [dispatch, genresList]);
+
   return (
     <StyledSection horizontal={horizontal}>
       {genres && genres?.map((genre) =>
```

Comment:
Tutaj bym się już bardziej nad tym zastanowił, bo niestety przechodząc do filmów, to mamy dispatchowanie akcji fetchGenres tyle razy ile mamy kafelek filmów :( To lepiej chyba byłoby już przenieść tego useEffecta do miejsc gdzie wyświetlają się listy filmów czyli PopularMovies i w PersonDetails ;)

Reply:
dispatch tutaj wykonuje się tylko jeśli lista gatunków jest pusta więc wykonałby się tylko raz, a tak tylko sprawdzał czy lista na pewno jest zapełniona. 
Przeniosę to do App.js, pamiętam że był kiedyś taki pomysł ale nie wiem czemu nie został wdrożony i pobieranie gatuków cały czas było w moviesPage i to jest problem jeśli nie zaczniesz wyświetlać strony od movies tylko wejdziesz od razu w person to gatunki nie zostaną pobrane, a są tam potrzebne.
Może być też tak jak mówisz w PopularMovies i w PersonDetails, ale najwięcej sensu ma App.js.

Reply:
Ja to nawet byłem pewien, że to już jest w App bo przy okazji wydzielenia genres do osobnego slice chyba była taka konkluzja żeby tak zrobić :) 

Tak czy owak, tak jak @katkowa prawi, do App zdecydowanie :)

Reply:
Aaaa 😄 ja tak samo jak @TSypko byłem już pewien, że to jest w App.js bo chyba sam ten pomysł proponowałem nawet 🤣